lib/autokey/service.py

```diff
@@ -519,17 +519,20 @@ def _record_error(self, script: typing.Union[model.Script, pathlib.Path], start_
         self.app.notify_error(error_record)
 
     def _execute(self, scope, script: typing.Union[model.Script, pathlib.Path]):
-        script_code, script_name = self._get_script_source_code_and_name(script)
-        logger.debug("About to execute script {}".format(script_name))
         start_time = datetime.datetime.now().time()
         # noinspection PyBroadException
-
         try:
-            compiled_code = compile(script_code, script_name, 'exec')
+            compiled_code = self._compile_script(script)
             exec(compiled_code, scope)
```

Comment:
The alternative, naming it `_compile_script_and_execute()` and moving the exec() inside the function, reduces one additional line.
But doing so will make the user-facing Traceback unnecessarily verbose, by adding another step. I’d like to avoid exposing too many internals in those Tracebacks, as those only clutter the data the user is actually interested in.

Reply:
That's a very good point.

Reply:
I guess ideally we'd modify the stack trace to only include user script errors here. I don't know how feasible that is, and even typing out "modify the stack trace" makes me feel uneasy…

Reply:
I think that would be ideal. But considering this is about running user scripts, I'm wondering how many users run other people's scripts?

Reply:
> I guess ideally we'd modify the stack trace to only include user script errors here. I don't know how feasible that is, and even typing out "modify the stack trace" makes me feel uneasy…

Hmmm. You’d have to dynamically analyse the traceback and remove all line tuples where the file name matches `__file__`. To determine is how that works if AK is ZIP archive packaged.

I’d like to avoid going that route, as that starts to obstruct the shown logic flow, in case someone does advanced stuff.

Reply:
Developers might run other people's scripts fairly often when they are submitted to recreate issues. And the "additional" detail (that users wouldn't normally see) would presumably be useful to developers running their own test scripts and phrases during the normal development/testing process. 

Also, as Thomas mentions, an advanced user might need the extra detail - but as an exception - not the rule.